### PR TITLE
📝 docs: Add pre-defined constraints in documentation

### PR DESCRIPTION
**Pull request by rlemaitre**
As I always struggled to remember pre-defined constraints, I added them to the mini-site.

### CODE REVIEW

File: docs/_docs/reference/constraint.md [hunk 1]
```diff
@@ -188,3 +188,104 @@ trait RefinedTypeOps[A, C, T]:
   def option(value: A): Option[T] =
     Option.when(rtc.test(value))(value.asInstanceOf[T])
 ```
+
+# Pre-defined constraints
```
Comment (Iltotore) [inline docs/_docs/reference/constraint.md:192]: This section (and children) should have an extra `#` as a `h1` (single `#`) is used for the title.

File: docs/_docs/reference/constraint.md [hunk 2]
```diff
@@ -188,3 +188,104 @@ trait RefinedTypeOps[A, C, T]:
...
+
+Iron provides a set of pre-defined constraints in the `io.github.iltotore.iron.constraint` package.
+You can find them in the [API documentation](https://iltotore.github.io/iron/api/io/github/iltotore/iron/constraint/index.html).
+
+## Global constraints
+
+Some constraints are available for all types.
+They are located in the `io.github.iltotore.iron.constraint.any` object.
```
Comment (Iltotore) [inline docs/_docs/reference/constraint.md:200]: What about using `[[package.name]]` instead of backquotes to link it to the Scaladoc?

**Review by Iltotore — CHANGES_REQUESTED** [review]
The list of pre-defined constraints is available in the [Scaladoc](https://iltotore.github.io/iron/io/github/iltotore/iron/constraint.html). Do you find this page not helpful enough?

Nonetheless, this PR looks with just some minor changes to make.

File: docs/_docs/reference/constraint.md (hunk 2 above)
Comment (rlemaitre) [reply docs/_docs/reference/constraint.md:200]: You're right, i'm fixing this

File: docs/_docs/reference/constraint.md (hunk 1 above)
Comment (rlemaitre) [reply docs/_docs/reference/constraint.md:192]: Fixed in the next commit

**Comment by rlemaitre** [comment]
> Do you find this page not helpful enough?

It's helpful, but I thought having all the constants in a single document may be clearer for some persons (including me 😃)

**Review by Iltotore — APPROVED** [review]

**Pull request merged**